x/concentrated-liquidity/pool.go

```diff
@@ -418,3 +420,97 @@ func (k Keeper) GetUserUnbondingPositions(ctx sdk.Context, address sdk.AccAddres
 	}
 	return userPositionsWithPeriodLocks, nil
 }
+
+// getPositionIDsByPoolID returns all position IDs for a given pool ID.
+func (k Keeper) getPositionIDsByPoolID(ctx sdk.Context, poolID uint64) ([]uint64, error) {
```

Comment:
seems like a useful query to expose in the future.

Reply:
WIP in separate PR